pkg/metrics/factory.go

```diff
@@ -16,7 +16,11 @@
 package metrics
```

Comment:
`metrics` is a package that defines the API. You're creating an implementation of that API, so place it in a different package, e.g. `pkg/metrics/otelmetrics`

Reply:
Noted!